heliopy/data/mms.py

```diff
@@ -7,131 +7,1781 @@
+        # depends on start[i].
+        trange = [None]*nfiles
+        for i in range(nfiles-1, -1, -1):
+            # Start tiem of interval
+            if len(parts[i][5]) == 8:
+                start = dt.datetime.strptime(parts[i][5], '%Y%m%d')
+            else:
+                start = dt.datetime.strptime(parts[i][5], '%Y%m%d%H%M%S')
+
+            # End time of interval. Subtract one second to prevent the file
+            # that begins at trange[i+1] is not included in any results.
+            if i == (nfiles-1):
+                end = self.end_date
+            elif i >= 0:
+                end = trange[i+1].start - dt.timedelta(seconds=1)
+
+            trange[i] = sunpy.time.TimeRange(start, end)
+
+        return trange
+
+    def local_dir(self, interval=None):
+        """
+        Local directory for a given interval. The interval should correspond
+        to a single data file (i.e. a single interval returned by
+        self.intervals).
+
+        Parameters
+        ----------
+        interval : sunpy.time.TimeRange
+
+        Returns
+        -------
+        dir : pathlib.Path
+            Local directory
+        """
+
+        # Create a list of sequential dates spanning the time interval
+        date = interval.start.to_datetime().strftime('%Y%m%d')
+
+        # Create the local directories
+        dir = construct_path(self.sc, self.instr, self.mode, self.level, date,
+                             optdesc=self.optdesc, root=self.data_root)
+        return dir[0]
+
+    def fname(self, interval, mirror=False):
+        """
+        Return the filename for a given interval from the SDC.
+
+        Parameters
+        ----------
+        interval : sunpy.time.TimeRange
+
+        Returns
+        -------
+        fname : str
+            Filename
+        """
+
+        # File names have version numbers in them. To get the most recent
+        # version, we must go through the SDC. Most functions key off the
+        # larger time interval of interest, so we must temporarily set it
+        # to the interval of a single data file.
+        interval_in = self.get_interval()
+        self.set_interval(interval)
+
+        # Get the file name
+        try:
+            file = self.fnames()
+            file = filter_time(file,
+                               interval.start.to_datetime(),
+                               interval.end.to_datetime()
+                               )
+
+            # Filter to within given interval.
+            if len(file) == 1:
+                file = file[0]
+            elif len(file) > 1:
+                ValueError('More than one file found. '
+                           'Restrict interval or use fnames().')
+        except IndexError:
+            file = ''
+
+        # Reset the time interval
+        self.set_interval(interval_in)
+
+        return file
+
+    @staticmethod
+    def datetime2interval(start_date, end_date):
```

Comment:
I'm not sure it's worth adding a whole method when this is just one line (`sunpy.time.TimeRange(start_date, end_date)`). I think the one line should be used everywhere and this method removed.